### PR TITLE
vapoursynth-editor: init at R19

### PR DESCRIPTION
**Pull request by tadeokondrak**
###### Motivation for this change

VapourSynth isn't very useful standalone without an editor.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

File: pkgs/development/libraries/vapoursynth/editor.nix [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    owner = "mystery_keeper";
+    repo = pname;
+    rev = stdenv.lib.toLower version;
+    sha256 = "1zlaynkkvizf128ln50yvzz3b764f5a0yryp6993s9fkwa7djb6n";
+  };
+
+  nativeBuildInputs = [ qmake makeWrapper ];
+  buildInputs = [ qtbase vapoursynth qtwebsockets ];
+
+  preConfigure = "cd pro";
```
Comment (teto) [inline pkgs/development/libraries/vapoursynth/editor.nix]: better use `sourceRoot = "pro";` so that one can override preConfigure later.

Comment (tadeokondrak) [reply pkgs/development/libraries/vapoursynth/editor.nix]: This doesn't work since it accesses files below the `pro/` folder while compiling.

Comment (teto) [reply pkgs/development/libraries/vapoursynth/editor.nix]: ok

**Comment by teto** [comment]
thanks !

**Pull request merged**